protocol/messenger_communities.go

```diff
@@ -463,3 +464,55 @@ func (m *Messenger) RemoveUserFromCommunity(id types.HexBytes, pkString string)
 	response.AddCommunity(community)
 	return response, nil
 }
+
+// RequestCommunityInfoFromMailserver installs filter for community and requests its details
+// from mailserver. When response received it will be passed through signals handler
+func (m *Messenger) RequestCommunityInfoFromMailserver(communityID string) error {
+
+	if _, ok := m.requestedCommunities[communityID]; ok {
+		return nil
+	}
+
+	//If filter wasn't installed we create it and remember for deinstalling after
+	//response received
+	filter := m.transport.FilterByChatID(communityID)
+	if filter == nil {
+		filters, err := m.transport.InitPublicFilters([]string{communityID})
+		if err != nil {
+			return fmt.Errorf("Can't install filter for community: %v", err)
+		}
+		if len(filters) != 1 {
+			return fmt.Errorf("Unexpected amount of filters created")
+		}
+		filter = filters[0]
+		m.requestedCommunities[communityID] = filter
+	} else {
+		//we don't remember filter id associated with community because it was already installed
+		m.requestedCommunities[communityID] = nil
+	}
+
+	_, err := m.RequestHistoricMessagesForFilter(context.Background(),
+		0,
+		0,
+		nil,
+		filter)
+
+	return err
+}
+
+// passCommunityInfoToSignalHandler calls signal handler with community info
+func (m *Messenger) passCommunityInfoToSignalHandler(communityID string) error {
+
+	if m.config.messengerSignalsHandler == nil {
+		return nil
+	}
+
+	//send signal to client that message status updated
+	community, err := m.communitiesManager.GetByIDString(communityID)
+	if err != nil || community == nil {
+		return fmt.Errorf("can't get community by ID: %v", err)
+	}
+
+	m.config.messengerSignalsHandler.CommunityInfoFound(community.Name(), community.Description(), community.MembersCount(), community.Verified())
```

Comment:
I would probably pass the whole community here, as we will likely show more data (the channels etc), so if you just pass the community object is going to be easier I think.